ui/app/pages/send/send.utils.js

```diff
@@ -205,20 +205,19 @@ async function estimateGas ({
   estimateGasMethod,
 }) {
   const paramsForGasEstimate = { from: selectedAddress, value, gasPrice }
-
   // if recipient has no code, gas is 21k max:
-  if (!sendToken && !data) {
+  if (Object.keys(sendToken).length === 0 && !data) {
     const code = Boolean(to) && await global.eth.getCode(to)
     // Geth will return '0x', and ganache-core v2.2.1 will return '0x0'
     const codeIsEmpty = !code || code === '0x' || code === '0x0'
     if (codeIsEmpty) {
       return SIMPLE_GAS_COST
     }
-  } else if (sendToken && !to) {
+  } else if (Object.keys(sendToken).length !== 0 && !to) {
     return BASE_TOKEN_GAS_COST
   }
 
-  if (sendToken) {
+  if (Object.keys(sendToken).length !== 0) {
```

Comment:
Though if `sendToken` is `undefined` or `null` this will throw now 🤔

Reply:
I suspect that this PR would fix the bug, but only because `estimateGas` is never called with `sendToken` set to `null` or `undefined`.

It's only called by `updateGasData` in `actions.js`, which itself is only called by `updateAndSetGasLimit` in `send.container.js`, which is only called in one spot in `send.component.js`, where `sendToken` is given a default value of `{}`: https://github.com/MetaMask/metamask-extension/blob/2873053d45f9769e77bc88833b9ba1b5d1bd7bde/ui/app/pages/send/send.component.js#L263

`state.metamask.send.token` is always set to `undefined` when there is no `sendToken`, so that default should always be set. But only for a very tenuous reason.

It would seem better to remove the `= {}` instead.